cpp/src/io/json/write_json.cu

```diff
@@ -417,21 +417,28 @@ struct column_to_strings_fn {
     auto child_view            = lists_column_view(column).get_sliced_child(stream_);
     auto constexpr child_index = lists_column_view::child_column_index;
     auto list_string           = [&]() {
-      auto child_string = [&]() {
-        if (child_view.type().id() == type_id::STRUCT) {
-          return (*this).template operator()<cudf::struct_view>(
-            child_view,
-            children_names.size() > child_index ? children_names[child_index].children
-                                                          : std::vector<column_name_info>{});
-        } else if (child_view.type().id() == type_id::LIST) {
-          return (*this).template operator()<cudf::list_view>(
-            child_view,
-            children_names.size() > child_index ? children_names[child_index].children
-                                                          : std::vector<column_name_info>{});
-        } else {
-          return cudf::type_dispatcher(child_view.type(), *this, child_view);
-        }
-      }();
+      // nulls are replaced due to special handling of all-null lists as empty lists
+      // by join_list_elements
+      auto child_string = cudf::strings::detail::replace_nulls(
```

Comment:
Non-blocker: I think I'd prefer this to be two separate lines. First the IIFE, then call `cudf::strings::detail::replace_nulls`. It's just pretty long and complicated to read. After 20 lines of code, I forget that we're computing a function argument.

Reply:
I used this to avoid the temporary. Updated.